apps/src/code-studio/components/progress/LessonLockDialog.jsx

```diff
@@ -11,264 +10,287 @@ import SectionSelector from './SectionSelector';
 import {teacherDashboardUrl} from '@cdo/apps/templates/teacherDashboard/urlHelpers';
 import {NO_SECTION} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
 
-class LessonLockDialog extends React.Component {
-  static propTypes = {
-    isOpen: PropTypes.bool.isRequired,
-    handleClose: PropTypes.func.isRequired,
-    initialLockStatus: PropTypes.arrayOf(
-      PropTypes.shape({
-        name: PropTypes.string.isRequired,
-        lockStatus: PropTypes.oneOf(Object.values(LockStatus)).isRequired
-      })
-    ),
-    selectedSectionId: PropTypes.number,
-    saving: PropTypes.bool.isRequired,
-    saveDialog: PropTypes.func.isRequired
-  };
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      lockStatus: props.initialLockStatus
-    };
-  }
+function LessonLockDialog({
+  isOpen,
+  handleClose,
+  initialLockStatus,
+  selectedSectionId,
+  saving,
+  saveDialog
+}) {
+  const [lockStatuses, setlockStatuses] = useState(initialLockStatus);
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if (nextProps.saving) {
-      return;
+  useEffect(() => {
+    if (!saving) {
+      setlockStatuses(initialLockStatus);
     }
-    this.setState({
-      lockStatus: nextProps.initialLockStatus
-    });
-  }
+  }, [initialLockStatus]);
 
-  setAllLockStatus(lockStatus) {
-    this.setState({
-      lockStatus: this.state.lockStatus.map(item =>
-        Object.assign({}, item, {
-          lockStatus
-        })
-      )
-    });
-  }
+  //
+  // Event handlers
+  //
+  const setAllLockStatus = lockStatus => {
+    setlockStatuses(currentLockStatuses =>
+      currentLockStatuses.map(item => ({...item, lockStatus}))
+    );
+  };
 
-  allowEditing = () => this.setAllLockStatus(LockStatus.Editable);
+  const allowEditing = () => setAllLockStatus(LockStatus.Editable);
 
-  lockLesson = () => this.setAllLockStatus(LockStatus.Locked);
+  const lockLesson = () => setAllLockStatus(LockStatus.Locked);
 
-  showAnswers = () => this.setAllLockStatus(LockStatus.ReadonlyAnswers);
+  const showAnswers = () => setAllLockStatus(LockStatus.ReadonlyAnswers);
 
-  viewSection = () => {
+  const viewSection = () => {
     const assessmentsUrl = teacherDashboardUrl(
-      this.props.selectedSectionId,
+      selectedSectionId,
       '/assessments'
     );
     window.open(assessmentsUrl, '_blank', 'noopener,noreferrer');
   };
 
-  handleRadioChange = event => {
+  const handleRadioChange = event => {
     const modifiedIndex = parseInt(event.target.name, 10);
     const value = event.target.value;
 
-    // Do this in a setTimeout so that the event has time to finish processing
-    // (i.e. so that checkbox is updated) before we call setState and rerender
```

Comment:
Were you able to verify that the `setTimeout` is not needed?

Reply:
Good question!  I tested this manually and the setTimeout didn't seem to be needed.  Have you seen us do something like the setTimeout in other parts of the code?

Reply:
No I haven't seen this before